### PR TITLE
update depdendencies and pom.xml to enable compilation with Java 11.

### PR DESCRIPTION
**Pull request by aaray**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -187,25 +191,25 @@
...
 			</plugin>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-compiler-plugin</artifactId>
-				<version>3.0</version>
+				<version>3.8.1</version>
 				<configuration>
-					<source>1.5</source>
-					<target>1.5</target>
+					<source>7</source>
```
Comment (ptomli) [inline pom.xml:204]: Switching away from Java 1.5 compat?

**Comment by aaray** [comment]
Compiling with recent versions of maven produces this error:

        [ERROR] Source option 5 is no longer supported. Use 7 or later.

Given that Java 5 was end-of-life back in October of 2009. I don´t see much of an issue with this update.

**Comment by ptomli** [comment]
@paoloc0 I guess this involves a new release. A bit outside my scope these days ;-)

**Comment by aaray** [comment]
Fair enough. Iĺl try to minimize the changes so that we can at least fix the build status on Travis-ci which has been broken for a while.

**Pull request closed without merging**